### PR TITLE
[MRG] Add sourmash logo to docs and README

### LINKED ISSUE #673: Design a logo for sourmash
We still don't have a logo for sourmash... And I'm jealous of [GROOT](https://github.com/will-rowe/groot/) and [HULK](https://github.com/will-rowe/hulk/), so let's fix this? =]

@taylorreiter and Clara Barcelos had a brainstorm and came up with this cute bucket, since sour mash is part of whisky brewing:
sourmash_bucket

Another approach would be to mix the brewing metaphor with... puppies? We are in a vet school, and a brewing puppy sounds cool.

We can also throw some other elements from use cases we explore (many datasets, taxonomic profiling, metagenomes...)

### PR DESCRIPTION
**Pull request by luizirber**
Fixes #673 

I also removed the customized template for the docs, opting for using the one that comes by default with Alabaster. It is a bit different (for example, "Navigation" instead of "Table of Contents" in the sidebar), but this way we don't need to keep updating our copy when the one from the theme is updated.

## TODO:

- [x] fix logo URL in README before merge
- [x] Credit the logo author (Stéfanie)

## Checklist

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
+[![Bioconda install](https://img.shields.io/conda/dn/bioconda/sourmash.svg?style=flag&label=Bioconda)](https://anaconda.org/bioconda/sourmash)
 <a href="https://pypi.org/project/sourmash/"><img alt="PyPI" src="https://badge.fury.io/py/sourmash.svg"></a>
 [![codecov](https://codecov.io/gh/dib-lab/sourmash/branch/master/graph/badge.svg)](https://codecov.io/gh/dib-lab/sourmash)
 [![DOI](http://joss.theoj.org/papers/10.21105/joss.00027/status.svg)](http://joss.theoj.org/papers/10.21105/joss.00027)
 <a href="https://github.com/dib-lab/sourmash/blob/master/LICENSE"><img alt="License: 3-Clause BSD" src="https://img.shields.io/badge/License-BSD%203--Clause-blue.svg"></a>
 
----
-
-Quickly search, compare, and analyze genomic and metagenomic data sets.
+<p align="center"><img src="https://raw.githubusercontent.com/dib-lab/sourmash/logo/doc/_static/logo.png" height="256" /></p>
```
Comment (luizirber) [inline README.md:13]: Need to fix this link before merge (point to `master`, not the `logo` branch), but this way it is possible to see how it looks.

**Comment by luizirber** [comment]
Before:
image
After:
image

**Comment by ctb** [comment]
👍

**Comment by luizirber** [comment]
Ready for review and merge @ctb @taylorreiter

**Pull request merged**